test/cluster/object_store/test_backup.py

```diff
@@ -587,26 +587,30 @@ async def create_cluster(topology, rf_rack_valid_keyspaces, manager, logger, obj
 
     return servers,host_ids
 
-def create_dataset(manager, ks, cf, topology, logger, dcs_num=None):
+async def create_dataset(manager, ks, cf, topology, logger, rf_across_dc_boundary=False, num_keys=256, min_tablet_count=None):
     cql = manager.get_cql()
-    logger.info(f'Create keyspace, rf={topology.rf}')
-    keys = range(256)
+    logger.info(f'Create keyspace, {topology=}')
+    keys = range(num_keys)
     replication_opts = {'class': 'NetworkTopologyStrategy'}
-    if dcs_num is not None:
-        for dc in range(dcs_num):
-            replication_opts[f'dc{dc}'] = int(topology.rf / dcs_num)
+    if rf_across_dc_boundary:
+        assert topology.rf % topology.dcs == 0, "rf must be divisible by number of dcs for rf_across_dc_boundary"
+        for dc in range(topology.dcs):
+            replication_opts[f'dc{dc}'] = int(topology.rf / topology.dcs)
     else:
         replication_opts['replication_factor'] = f'{topology.rf}'
+    replication_opts['replication_factor'] = f'{topology.rf}'
```

Comment:
This statement is currently only in the `else:` block.
Why repeat it on the common path?